core/src/main/java/com/google/instrumentation/trace/Trace.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.instrumentation.common.Provider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Class that manages a global instance of the {@link TraceService}. */
```

Comment:
are objects returned here safe to cache? If so, maybe note that.

Also, static singletons is to some an anti-pattern reserved for edge cases where there's no other option. Is that the thinking here? a "last resort" or are you expecting tracing users to instantiate a JVM singleton normally? Either way, might be worthwhile documenting the intended use.